StRoot/StSpinPool/StJetSkimEvent/StJetSkimEvent.h

```diff
@@ -145,357 +145,397 @@ inline void StJetSkimTrig::setTotalEnergy(int aEnergy) {mTotalEnergy = aEnergy;}
+
+  void setNTzero(short n) { mNTzero = n; }
+  void setNTzeroCan(short nCan) { mNTzeroCan = nCan; }
+  void setTCanFirst(double tFirst) { mTCanFirst = tFirst; }
+  void setTCanLast(double tLast) { mTCanLast = tLast; }
+  void setNVpdEGoodHits(int vpdEGoodHits) { mNVpdEGoodHits = vpdEGoodHits; }
+  void setNVpdWGoodHits(int vpdWGoodHits) { mNVpdWGoodHits = vpdWGoodHits; }
+  void setEarliestVpdEHit(int earliestVpdEHit) { mEarliestVpdEHit = earliestVpdEHit; }
+  void setEarliestVpdWHit(double earliestVpdWHit) { mEarliestVpdWHit = earliestVpdWHit; }
+  void setClosestVpdEHit(double closestVpdEHit) { mClosestVpdEHit = closestVpdEHit; }
+  void setClosestVpdWHit(double closestVpdWHit) { mClosestVpdWHit = closestVpdWHit; }
+  void setLatestVpdEHit(double latestVpdEHit) { mLatestVpdEHit = latestVpdEHit; }
+  void setLatestVpdWHit(double latestVpdWHit) { mLatestVpdWHit = latestVpdWHit; }
+
+  //gets
+  float fill() const {return mFill;}
+  int runId() const {return mRunId;}
+  int eventId() const {return mEventId;}
+  TObjString mudstFileName() const {return mMudstFileName;}
+  
+  int bx7() const {return mbx7;}
+  int bx48() const {return mbx48;}
+  int spinBits() const {return mSpinBits;}
+
+  int eBbc() const {return mEbbc;}
+  int wBbc() const {return mWbbc;}
+  int bbcTimeBin() const {return mBbcTimeBin;}
+
+  float vpdTdiff() const {return mVpdTdiff;}
+  float vpdTstart() const {return mVpdTstart;}
+  float vpdZvertex() const {return mVpdZvertex;}
+  int vpdEastHits() const {return mVpdEastHits;}
+  int vpdWestHits() const {return mVpdWestHits;}
+
+  float zdcWestRate() const { return mZdcWestRate; }
+  float zdcEastRate() const { return mZdcEastRate; }
+  float zdcCoincidenceRate() const { return mZdcCoincidenceRate; }
+  float bbcWestRate() const { return mBbcWestRate; }
+  float bbcEastRate() const { return mBbcEastRate; }
+  float bbcCoincidenceRate() const { return mBbcCoincidenceRate; }
+
+  int isValid() const {return mIsValid;}
+  int isPolLong() const {return mIsPolLong;}
+  int isPolTrans() const {return mIsPolTrans;}
+  int isMaskedUsingBx48() const {return mIsMaskedUsingBx48;}
+  int offsetBx48minusBX7() const {return mOffsetBx48minusBX7;}
+  int spin4usingBx48() const {return mSpin4usingBx48;}
+
+  const TClonesArray* triggers() const {return mTriggers;}
+  TClonesArray* triggers() {return mTriggers;}
+  StJetSkimTrig* trigger(int trigId);
+
+  TClonesArray* trigHeaders() {return (TClonesArray*)mTrigHeaderArrayRef.GetObject();}
+  StJetSkimTrigHeader* trigHeader(int trigId);
+  
+  const TClonesArray* vertices() const {return mVertices;}
+  StJetSkimVert* bestVert() const;
+  
+  const int* L2Result() const { return mL2Result; }
+  
+  const StPythiaEvent* mcEvent() const {return mMcEvent;}
+
+  short nTzero() const { return mNTzero; }
+  short nTzeroCan() const { return mNTzeroCan; }
+  float tStart() const {return mVpdTstart;}
+  double tCanFirst() const { return mTCanFirst; }
+  double tCanLast() const { return mTCanLast; }
```

Comment:
```suggestion
  short nTzero() const { return mNTzero; } //! Get number of particles used for start-time calculations
  short nTzeroCan() const { return mNTzeroCan; } //! Get number of candidate particles available for start-time calculations before outlier rejection
  double tCanFirst() const { return mTCanFirst; } //! Get earliest candidate particle in start-time calculations
  double tCanLast() const { return mTCanLast; } //! Get latest candidate particle in start-time calculations
```

The `tStart()` is a duplicate of `vpdTstart()` above

Reply:
@veprbl Carl and I wanted to leave the vpdTstart() function alone and not change it. However, the start-time returned by the vpdTstart() function in not necessarily always the start-time from the VPD. So, we added a start-time function that has a more generic name where there is no reference to what detector (VPD or BTOF) calculated the start-time.

Reply:
This may be premature to add such function before the different behaviours are actually implemented, but okay. Can we, please, add a comment to explain your reasoning in the code?

Reply:
@veprbl Yeah! It is totally fine with me to add a comment explaining our reasoning.

Reply:
```suggestion
  //! This method returns a generic start-time. Unlike
  //! StJetSkimEvent::vpdTstart it does not reference which detector
  //! (VPD or BTOF) is used calculated the start-time.
  float tStart() const {return mVpdTstart;}

  short nTzero() const { return mNTzero; } //! Get number of particles used for start-time calculations
  short nTzeroCan() const { return mNTzeroCan; } //! Get number of candidate particles available for start-time calculations before outlier rejection
  double tCanFirst() const { return mTCanFirst; } //! Get earliest candidate particle in start-time calculations
  double tCanLast() const { return mTCanLast; } //! Get latest candidate particle in start-time calculations
```

Reply:
@veprbl @BassamAboona, what about we removed the tStart() function, instead I added a comment saying vpdTStart() isn't necessarily coming from VPD? Two reasons I think we should remove this function: a) there isn't a set function corresponding to tStart(); b) I also checked the StJetSkimEventMaker, in that function the mVpdTstart is also set by setVpdTstart(). To make code consistent overall, removing tStart() may be a better option. In the analysis code, you can always note what this time is for. By the way, can this time come from start-less tof?

Reply:
> vpdTStart() isn't necessarily coming from VPD

If it's named "vpdTStart", it better come from VPD. If not, it's an unfortunate bug.

Reply:
@veprbl and @zlchang Yeah! This "VPD" start-time can come from start-less tof. I think historically, start-less TOF start-time was not widely used by the Spin group because the algorithm for start-less TOF was not optimized for our collision energies. So, it used to be safe to assume that the start-time was primarily coming from the VPD, and thus the name "vpdTStart()" for the function. However, with the recent work that Carl and I did, start-less TOF can provide start-time a lot more often than it used. However, an event can either have a start-time calculated by the VPD or start-less TOF and not both.

I am okay with removing the tStart() function, but users need to know, and actually dig in the code and read the comments, that "vpdTStart" doesn't mean that the start-time necessarily came from the VPD.

Alternatively, the "vpdTStart" can be renamed "tStart"

Reply:
@BassamAboona I'm not aware of this. Could you please point me to where btofHeader is mangled? Is there a some kind of startless tof maker that replaces the vpd time?

Reply:
@veprbl The process of finding the start-time is mostly done in StBTofCalibMaker and the start-time is stored in StBTofHeader. I think the best thing to do is search "mTStart" in StBTofCalibMaker. You also might want to look at the routine in there named tstart_NoVpd() on line 2050. 

StRoot/StBTofCalibMaker/StBTofCalibMaker.cxx

Reply:
@BassamAboona do you know if there is a function to return VPD time in the btofheader? If not, we can rename the variable, the setter function and the getter function.

Reply:
@zlchang There is no function in StBTofHeader that specifically returns VPD start-time. StBTofHeader has a function named tStart(). This will return the event's start-time and the returned start-time could have been calculated using start-less TOF, VPD, etc. Carl and I intended to mimic this function in the jet codes because "vpdTStart()" is misleading, but we also did not want to just modify the existing "vpdTStart()". 

In other words, if an event has a start-time, then it is stored in the same spot in StBTofHeader regardless of what method was used to obtain the start-time. One event can not have two start-times that are calculated using different methods, which then makes "vpdTStart()" misleading if the start-time was actually calculated using start-less TOF.

Reply:
> @BassamAboona do you know if there is a function to return VPD time in the btofheader? If not, we can rename the variable, the setter function and the getter function.

It seems like it's there isn't. It's either `StBTofCalibMaker::tstart` or `StBTofCalibMaker::tstart_NoVpd` and the later ignores VPD time information from `mTSumEast`/`mTSumWest`. I think we should just get rid of the `StJetSkimEvent::vpdTStart()`.

Reply:
@veprbl Right it is either start-less TOF or VPD start-time and not both.